renderers/chords_renderer.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const parseVerse = require('../verse.js')['parseVerse'];
+
+class ChordsRenderer {
+  constructor(chordChartJson) {
+    this.chordChartJson = chordChartJson;
+  }
+
+  createHtmlChordChart() {
+    const chordChartHtml = document.createElement('div');
+    chordChartHtml.className = 'chart';
+
+    this.chordChartJson.forEach((verse) => {
+      const verseDiv = document.createElement('div');
+      verseDiv.className = 'verse';
+
+      this.createListOfWrappedVoices(verse)
+        .forEach((event) => {
+          verseDiv.appendChild(event);
+        });
+
+      chordChartHtml.appendChild(verseDiv);
+    });
+
+    return chordChartHtml;
+  }
+
+  createListOfWrappedVoices(verse) {
+    // Assumes voices are sorted by index and grouped by voice.
+    const voiceElements = new Map();
+
+    verse.forEach((event) => {
+      if (!voiceElements.has(event.voice)) {
+        voiceElements.set(event.voice, new Map([['index', 0], ['parentElement', document.createElement('div')]]));
+
+        voiceElements.get(event.voice).get('parentElement').className = event.voice;
+      }
+
+      const voice = voiceElements.get(event.voice);
+      this.appendVoiceContentDiv(voice.get('parentElement'), event.content, event.index - voice.get('index'));
+
+      voice.set('index', event.index + event.content.toString().length);
+    });
+
+    // Voices should have an order. The chords should be first, followed by all other voices.
```

Comment:
voiceElements is a map. I'm not sure that this is guaranteed to be true. Unless maps keep the order that keys were set? In which case, it's only true by luck. I didn't enforce the sorting of the voices in the phrase to have chords first. It just so happens that everything we parse has chords first. Let's add a TODO and create an issue to revisit this.

Reply:
`Map` in javascript does actually maintain an order. It's the insertion order of keys.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared

Bad comment, I'll try to clarify what I meant. While parsing a verse, whatever order the lines were written is maintained. For example, in this phrase, `c1` voice will be parsed and inserted to the map before `v1` voice.

```chords
c1:                    Am     G  F          G      Esus4  E
v1: All the leaves are brown        and the sky is gray
```

If for some reason, the markdown had chords at the bottom, I think it should be expected that's how it's displayed.

```chords
v1: All the leaves are brown        and the sky is gray
c1:                    Am     G  F          G      Esus4  E
```